src/lighteval/tasks/templates/nli.py

```diff
@@ -0,0 +1,221 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Callable, Literal
+
+from typing_extensions import NotRequired, TypedDict
+
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.templates.multichoice import get_mcq_prompt_function
+from lighteval.tasks.templates.utils.formatting_utils import PUNCT, capitalize, decapitalize, fix_ending_punct
+from lighteval.tasks.templates.utils.formulation import CFFormulation, Formulation, HybridFormulation, MCFFormulation
+from lighteval.tasks.templates.utils.translation_literals import TRANSLATION_LITERALS, TranslationLiterals
+from lighteval.tasks.templates.utils.utils import create_adapter_from_dict
+from lighteval.utils.language import Language
+
+
+NLI_TEMPLATE_QUERY_CF = "{instruction}{premise}{word_space}{confirmation_word}{question_mark}"
+NLI_TEMPLATE_CONT_CF = "{sentence_space}{label}{comma}{word_space}{hypothesis}"
+
+
+class NLIInput(TypedDict):
+    premise: str
+    hypothesis: str
+    gold_idx: int
+    instruction: NotRequired[str]
+
+
+class NLIAdapter(TypedDict):
+    premise: str
+    hypothesis: str
+    gold_idx: str
+    instruction: NotRequired[str]
+
+
+RelationType = Literal["entailment", "neutral", "contradiction"]
+
+
+def get_nli_prompt_function_natural(
+    language: Language,
+    adapter: Callable[[dict], NLIInput] | NLIAdapter,
+    relations: list[RelationType],
+):
+    """
+    Create a templated prompt function for a Natural Language Inference (NLI) task.
+    Example tasks:
+    - ANLI
+    - XNLI
+
+    Format:
+    Premies right? | (Yes,No,Also), hypothesis
+
+    Args:
+        language (Language): The language of the NLI task.
+        adapter (Callable[[dict], NLIInput] | NLIAdapter): A function or dictionary to adapt the input data to the required NLIInput format.
+            Must map data from the dataset row to the NLIInput format.
+            Note: The label must be a index in the relations list, based on the correct relation between the premise and hypothesis.
+
+        relations (list[Literal["entailment", "neutral", "contradiction"]]): The list of possible relations for the NLI task.
+        formulation (MCFFormulation, optional): The formulation to use for multiple-choice questions. Defaults to MCFFormulation().
+
+    Returns:
+        Callable: A function that generates NLI prompts based on the given parameters.
+    """
+
+    def get_relation_label(label: RelationType, translation_literals: TranslationLiterals):
+        if label == "entailment":
+            return translation_literals.yes
+        elif label == "contradiction":
+            return translation_literals.no
+        elif label == "neutral":
+            return translation_literals.also
```

Comment:
else raise error?

Reply:
That can't happen

Reply:
I would not underestimate the creativity of some users ^^ and add an error message - but program will fail later anyway so up to you